contrib/flo-styx/src/main/java/com/spotify/flo/contrib/styx/Environment.java

```diff
@@ -0,0 +1,172 @@
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.flo.contrib.styx;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Sanitized Styx environment variables where key should match "^([a-z]([-a-z0-9]*[a-z0-9])?)?$" and
+ * less than 64 characters, and value should match "^([a-z]([-a-z0-9]*[a-z0-9])?)?$" and less than
+ * 64 characters.
+ */
+public final class Environment {
+
+  private static final String DEFAULT_KEY_PREFIX = "spotify";
+
+  private static final String STYX_COMPONENT_ID = "styx.component.id";
+  private static final String STYX_WORKFLOW_ID = "styx.workflow.id";
+  private static final String STYX_PARAMETER = "styx.parameter";
+  private static final String STYX_EXECUTION_ID = "styx.execution.id";
+  private static final String STYX_TRIGGER_ID = "styx.trigger.id";
+  private static final String STYX_TRIGGER_TYPE = "styx.trigger.type";
+
+  private static final String STYX_COMPONENT_ID_KEY = "styx-component-id";
+  private static final String STYX_WORKFLOW_ID_KEY = "styx-workflow-id";
+  private static final String STYX_PARAMETER_KEY = "styx-parameter";
+  private static final String STYX_EXECUTION_ID_KEY = "styx-execution-id";
+  private static final String STYX_TRIGGER_ID_KEY = "styx-trigger-id";
+  private static final String STYX_TRIGGER_TYPE_KEY = "styx-trigger-type";
+
+  private static final int KEY_MAX_LENGTH = 63;
+  private static final int VALUE_MAX_LENGTH = 63;
+  private static final String KEY_REGEX = "^[a-z]([-a-z0-9]*[a-z0-9])?$";
+  private static final String VALUE_REGEX = "^([a-z]([-a-z0-9]*[a-z0-9])?)?$";
+  private static final String VALUE_REPLACE_REGEX = "[^-a-z0-9]";
+
+  private Environment() {
+    throw new UnsupportedOperationException();
+  }
+
+  /**
+   * Get sanitized Styx environment variables which are visible to containers launched by Styx:
+   *
+   * <ul>
+   * <li>spotify-styx-component-id</li>
+   * <li>spotify-styx-workflow-id</li>
+   * <li>spotify-styx-parameter</li>
+   * <li>spotify-styx-trigger-id</li>
+   * <li>spotify-styx-trigger-type</li>
+   * </ul>
+   *
+   * @return Styx environment variables which are visible to containers launched by Styx
+   */
+  public static Map<String, String> getEnv() {
+    return getEnv(DEFAULT_KEY_PREFIX);
+  }
```

Comment:
Defaulting to `spotify` here seems not too useful for people not in `spotify`.

If no prefix is set, shouldn't values just be e.g. `styx-component-id` rather than `-styx-component-id`?

Reply:
I guess we have been trying to avoid binding `flo-styx` with GCP stuff. As you pointed out, I really struggled how to name this and that's why I tried to be extra clear in javadoc. `getSanitizedEnv` seems to be something better.

Reply:
I don't believe such things makes life easier for anyone. Compare apollo, they seem to try to not have spotify defaults but make it easy to set what you need. We could then have our documentation open source too rather than internal.

Naming things `…Default` and then returning something with spotify in it is just unexpected.

Reply:
Do we really expect everyone writes their pipelines like this? If yes I'm fine.

````
dfOptions.setLabels(Environment.getSanitizedEnv("spotify"))
```

Reply:
What do we say? Remove the default?

Reply:
I guess we would eventually have an internal Beam `PipelineOptions` builder/factory which would encapsulate "Spotify defaults" (and additional validations if necessary), e.g. region, service account and labels. This is possibly outside of the scope of `flo` and more into how we use `Beam` and `scio`.

Reply:
That was attempted by https://github.com/spotify/flo/pull/76 as well as an internal PR. As you said, it's more beam/scio than flo, so we closed that PR.